demos/mqtt/mqtt_demo_mutual_auth/mqtt_demo_mutual_auth.c

```diff
@@ -976,11 +996,26 @@ static int establishMqttSession( MQTTContext_t * pMqttContext,
      * reestablish a session which was already present. */
     connectInfo.cleanSession = createCleanSession;
 
-    /* The client identifier is used to uniquely identify this MQTT client to
+    /* Generate a random number to prefix to the client ID string.
+     * The randomization of the client ID is used to avoid
+     * client identifier collisions when connecting to the MQTT broker. */
+    randomNumForClientId = ( rand() % ( MAX_RAND_NUMBER_FOR_CLIENT_ID + 1u ) );
```

Comment:
In the man page, it is mentioned that the lower bits of rand are less random than the higher bits on many platforms. This suggests collisions may be further degraded in different environments.